org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/preferences/OptionsConfigurationBlock.java

```diff
@@ -1,697 +1,698 @@
-	}
-
-	protected abstract Job getBuildJob(IProject project);
-
-	protected abstract String[] getFullBuildDialogStrings(boolean workspaceSettings);
-
-	public void performDefaults() {
-		for (int i = 0; i < keys.length; i++) {
-			String curr = keys[i];
-			setToDefault(curr);
-		}
-		updateControls();
-		validateSettings(null, null, null);
-	}
-
-	public void dispose() {
-	}
-
-	protected void updateControls() {
-		for (int i = comboBoxes.size() - 1; i >= 0; i--) {
-			updateCombo(comboBoxes.get(i));
-		}
-		for (int i = checkBoxes.size() - 1; i >= 0; i--) {
-			updateCheckBox(checkBoxes.get(i));
-		}
-		for (int i = textBoxes.size() - 1; i >= 0; i--) {
-			updateText(textBoxes.get(i));
-		}
-	}
-
-	protected void collectRegistredKeys() {
-		List<String> collectedKeys = Lists.newArrayList();
-		for (Combo combo : comboBoxes) {
-			ControlData data = (ControlData) combo.getData();
-			collectedKeys.add(data.getKey());
-		}
-		for (Button button : checkBoxes) {
-			ControlData data = (ControlData) button.getData();
-			collectedKeys.add(data.getKey());
-		}
-		for (Text textField : textBoxes) {
-			collectedKeys.add((String) textField.getData());
-		}
-		collectedKeys.addAll(keysToRegister);
-		this.keys = Iterables.toArray(collectedKeys, String.class);
-	}
-
-	protected void updateCombo(Combo curr) {
-		ControlData data = (ControlData) curr.getData();
-		String currValue = getValue(data.getKey());
-		int selection = data.getSelection(currValue);
-		curr.select(selection);
-	}
-
-	protected void updateCheckBox(Button curr) {
-		ControlData data = (ControlData) curr.getData();
-		curr.setSelection(checkBoxValue(data));
-	}
-
-	private boolean checkBoxValue(ControlData data) {
-		String currValue = getValue(data.getKey());
-		boolean selection = data.getSelection(currValue) == 0;
-		return selection;
-	}
-
-	protected void updateText(Text curr) {
-		String key = (String) curr.getData();
-		String currValue = getValue(key);
-		if (currValue != null) {
-			curr.setText(currValue);
-		}
-	}
-
-	protected abstract void validateSettings(String changedKey, String oldValue, String newValue);
-
-	private void captureOriginalSettings(String[] keys) {
-		originalSettings = Maps.newHashMapWithExpectedSize(keys.length);
-		for (String key : keys) {
-			originalSettings.put(key, preferenceStore.getString(key));
-		}
-	}
-
-	protected void registerKey(String key) {
-		keysToRegister.add(key);
-	}
-
-	public abstract String getPropertyPrefix();
-
-	public String getIsProjectSpecificPropertyKey(String propertyPrefix) {
-		String key = IS_PROJECT_SPECIFIC;
-		if (propertyPrefix != null) {
-			key = isProjectSpecificPropertyKey(propertyPrefix);
-		} else {
-			logError("Project specific key is not qualified", null);
-		}
-		return key;
-	}
-
-	public static String isProjectSpecificPropertyKey(String propertyPrefix) {
-		return propertyPrefix + "." + IS_PROJECT_SPECIFIC;
-	}
-}
+/*******************************************************************************
```

Comment:
There are so many changes in this file because its line-endings are not normalized.
I just created https://github.com/eclipse/xtext/pull/2977 to fix that for the entire repo and converted this to a draft.
Once https://github.com/eclipse/xtext/pull/2977 is decided I think this should be rebased/redone in order to minimize the changes.